scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_model.py

```diff
@@ -7,28 +7,28 @@
 from Muon.GUI.Common.utilities.algorithm_utils import run_Fit, run_simultaneous_Fit
 import mantid
 from Muon.GUI.Common.ADSHandler.muon_workspace_wrapper import MuonWorkspaceWrapper
-from Muon.GUI.Common.ADSHandler.workspace_naming import get_fit_workspace_base_directory
+from Muon.GUI.Common.ADSHandler.workspace_naming import get_fit_workspace_directory
 from mantid.simpleapi import RenameWorkspace
-from mantid.api import FunctionFactory
 
 
 class FittingTabModel(object):
     def __init__(self):
         pass
 
     def do_single_fit(self, parameter_dict):
-        output_workspace, fitting_parameters_table, function_string, output_status, output_chi_squared =\
+        group_name = parameter_dict.pop('GroupName')
+        output_workspace, fitting_parameters_table, function_string, output_status, output_chi_squared = \
             self.do_single_fit_and_return_workspace_parameters_and_fit_function(parameter_dict)
 
-        workspace_name, directory = self.create_fitted_workspace_name(parameter_dict['InputWorkspace'],
-                                                                      parameter_dict['Function'])
-        table_name, directory = self.create_parameter_table_name(parameter_dict['InputWorkspace'],
-                                                                 parameter_dict['Function'])
+        workspace_name, workspace_directory = self.create_fitted_workspace_name(parameter_dict['InputWorkspace'],
+                                                                                parameter_dict['Function'], group_name)
+        table_name, table_directory = self.create_parameter_table_name(parameter_dict['InputWorkspace'],
+                                                                       parameter_dict['Function'], group_name)
 
-        self.add_workspace_to_ADS(output_workspace, workspace_name, directory)
-        self.add_workspace_to_ADS(fitting_parameters_table, table_name, directory)
+        self.add_workspace_to_ADS(output_workspace, workspace_name, workspace_directory)
+        self.add_workspace_to_ADS(fitting_parameters_table, table_name, table_directory)
 
-        return function_string, output_status, output_chi_squared
+        return function_string.clone(), output_status, output_chi_squared
```

Comment:
I thinks its a function and not a function string, in which case we should update the name